tests/unit/aws/test_access.py

```diff
@@ -130,9 +126,10 @@ def test_create_external_locations(mock_ws, installation_multiple_roles, mock_aw
         call(mock.ANY, 's3://BUCKETX/FOLDERX', 'credx', skip_validation=True),
     ]
     mock_ws.external_locations.create.assert_has_calls(calls, any_order=True)
+    aws.get_role_policy.assert_not_called()
```

Comment:
this should still work, without re-adding `aws = create_autospec(AWSResources)`
```suggestion
    mock_aws.get_role_policy.assert_not_called()
```

Reply:
`mock_aws` was a bit difficult to read